### PR TITLE
Fix restler gb collection

### PR DESCRIPTION
**Pull request by stevensnicholas**
- Format with black (all python files in repo)
- Added a github action for a black lint cheaker
- moved print to logger 
- Alter /tmp/Compile/engine_settings.json to set engine_settings["garbage_collection_interval"] = 0
Otherwise reslter will attempt to use multiprocessing python libraries which are not supported in lambda.  See https://stackoverflow.com/questions/34005930/multiprocessing-semlock-is-not-implemented-when-running-on-aws-lambda
- Upload compile folder into s3 (helps debugging)
- convert upload folder to for loop with three paths (compile, logs, results)
- logs print s3 upload location

### CODE REVIEW

**Comment by The-TT-Hacker** [comment]
Can you add the black formatter to the pipeline?

File: restler/app/app.py [hunk 1]
```diff
@@ -5,9 +5,10 @@
...
 import shutil
 
+
 def handler(event, context):
-    #Restler consistently tries to save files to working directory
-    #So to avoid read only errors in lambda change wdir to /tmp
+    # Restler consistently tries to save files to working directory
+    # So to avoid read only errors in lambda change wdir to /tmp
     os.chdir("/tmp")
     print("logging started")
```
Comment (The-TT-Hacker) [inline restler/app/app.py:13]: if you can use the python logging library instead of print that would be good

Comment (The-TT-Hacker) [reply restler/app/app.py:13]: can be done in another PR

Comment (stevensnicholas) [reply restler/app/app.py:13]: Added in with new commit

**Review by thefishua — APPROVED** [review]
LGTM fixed issues with logging and s3 uploading

**Pull request merged**